File: 205-isomorphic-strings/205-isomorphic-strings.rs
use std::collections::{HashMap, HashSet};



impl Solution {
    pub fn is_isomorphic(s: String, t: String) -> bool {
        let s: Vec<char> = s.chars().collect();
        let t: Vec<char> = t.chars().collect();
        let mut chars = HashMap::new();
        
        for i in 0..s.len() {
            let v = chars.entry(s[i]).or_insert(t[i]);
            
            if *v != t[i] {
                return false;
            }
        }

        chars.keys().collect::<HashSet<_>>().len() == chars.values().collect::<HashSet<_>>().len()
    }
}

File: 205-isomorphic-strings/205-isomorphic-strings.ts
function isIsomorphic(s: string, t: string): boolean {
    const charMap = new Map();
    for (let i = 0; i < s.length; i++) {
        if(charMap.has(s[i]) && charMap.get(s[i]) != t[i]) {
            return false;
        }
        charMap.set(s[i], t[i])
    }
    return new Set(charMap.keys()).size == new Set(charMap.values()).size
};